nitransforms/tests/test_surface.py

```diff
@@ -0,0 +1,207 @@
+#         np.testing.assert_array_equal(xfm.mat.indices, xfm2.mat.indices)
+#         np.testing.assert_array_equal(xfm.mat.indptr, xfm2.mat.indptr)
+#     except Exception:
+#         os.remove(fn)
+#         raise
+#     os.remove(fn)
+
+
+# def test_surface_transform_normalization():
+#     mat = np.random.uniform(size=(20, 10))
+#     xfm = SurfaceCoordinateTransform(mat)
+#     x = np.random.uniform(size=(5, 20))
+#     y_element = xfm.apply(x, normalize="element")
+#     np.testing.assert_array_less(y_element.sum(axis=1), x.sum(axis=1))
+#     y_sum = xfm.apply(x, normalize="sum")
+#     np.testing.assert_allclose(y_sum.sum(axis=1), x.sum(axis=1))
+#     y_none = xfm.apply(x, normalize="none")
+#     assert y_none.sum() != y_element.sum()
+#     assert y_none.sum() != y_sum.sum()
+def test_SurfaceTransformBase(testdata_path):
+    # note these transformations are a bit of a weird use of surface transformation, but I'm
+    # just testing the base class and the io
+    sphere_reg_path = testdata_path / "sub-sid000005_ses-budapest_acq-MPRAGE_hemi-R_space-fsLR_desc-reg_sphere.surf.gii"
+    pial_path = testdata_path / "sub-sid000005_ses-budapest_acq-MPRAGE_hemi-R_pial.surf.gii"
+
+    sphere_reg = SurfaceMesh(nb.load(sphere_reg_path))
+    pial = SurfaceMesh(nb.load(pial_path))
+    stfb = SurfaceTransformBase(sphere_reg, pial)
+
+    # test loading from filenames
+    stfb_ff = SurfaceTransformBase.from_filename(sphere_reg_path, pial_path)
+    assert stfb_ff == stfb
+
+    # test inversion and setting
+    stfb_i = ~stfb
+    stfb.reference = pial
+    stfb.moving = sphere_reg
+    assert np.all(stfb_i._reference._coords == stfb._reference._coords)
+    assert np.all(stfb_i._reference._triangles == stfb._reference._triangles)
+    assert np.all(stfb_i._moving._coords == stfb._moving._coords)
+    assert np.all(stfb_i._moving._triangles == stfb._moving._triangles)
+    # test equality
+    assert stfb_i == stfb
+
+
+def test_SurfaceCoordinateTransform(testdata_path):
+    # note these transformations are a bit of a weird use of surface transformation, but I'm
+    # just testing the class and the io
+    sphere_reg_path = testdata_path / "sub-sid000005_ses-budapest_acq-MPRAGE_hemi-R_space-fsLR_desc-reg_sphere.surf.gii"
+    pial_path = testdata_path / "sub-sid000005_ses-budapest_acq-MPRAGE_hemi-R_pial.surf.gii"
+    fslr_sphere_path = testdata_path / "tpl-fsLR_hemi-R_den-32k_sphere.surf.gii"
+
+    sphere_reg = SurfaceMesh(nb.load(sphere_reg_path))
+    pial = SurfaceMesh(nb.load(pial_path))
+    fslr_sphere = SurfaceMesh(nb.load(fslr_sphere_path))
+
+    # test mesh correspondence test
+    with pytest.raises(ValueError):
+        sct = SurfaceCoordinateTransform(fslr_sphere, pial)
+
+    # test loading from filenames
+    sct = SurfaceCoordinateTransform(sphere_reg, pial)
+    sctf = SurfaceCoordinateTransform.from_filename(reference_path=sphere_reg_path,
+                                                    moving_path=pial_path)
```

Comment:
I believe this is reversed:
```suggestion
    sct = SurfaceCoordinateTransform(pial, sphere_reg)
    sctf = SurfaceCoordinateTransform.from_filename(reference_path=pial_path,
                                                    moving_path=sphere_reg_path)
```